lib/Runtime/Base/FunctionBody.h

```diff
@@ -1348,16 +1393,180 @@ namespace Js
         ScriptFunctionType * AllocDeferredPrototypeType();
     };
 
+    inline Js::LocalFunctionId FunctionProxy::GetLocalFunctionId() const
+    {
+        Assert(GetFunctionInfo());
+        return GetFunctionInfo()->GetLocalFunctionId();
+    }
+
+    inline void FunctionProxy::SetLocalFunctionId(LocalFunctionId functionId)
+    {
+        Assert(GetFunctionInfo());
+        Assert(GetFunctionInfo()->GetFunctionProxy() == this);
```

Comment:
Can these asserts be moved to GetFunctionInfo itself? Also was curious why this particular assert wasn't there in GetLocalFunctionId (another reason I suggested centralizing the asserts)